### DISCUSSION TITLE
current runner is not compatible with gauge LSP

### DISCUSSION THREAD

**Discussion by 2vireddy** [getgauge/gauge#1950] [General]
I am trying to setup a gauge python project and already have a framework which works on other machines of my colleagues but i am unable to get gauge up on my Windows PC . 

[Gauge] [INFO] Starting language runner
05-04-2021 20:18:16.472 [python] [INFO] Python: 3.8.2
05-04-2021 20:18:16.472 [python] [DEBUG] Loading step implementations from s:\Git-TVL-Mar25\TVLQA\gauge-pom\step_impl dirs.
05-04-2021 20:18:39.103 [Gauge] [DEBUG] Gauge could not initialize. Install 'ms-python.python' extension for code insights. For more information see[Problems](command:workbench.actions.view.problems), check logs.[Troubleshooting](https://docs.gauge.org/troubleshooting.html?language=javascript&ide=vscode#gauge-could-not-initialize-for-more-information-see-problems)
Some of the gauge lsp feature will not work as expected.
05-04-2021 20:18:39.103 [Gauge] [INFO] LangServer: reading on stdin, writing on stdout
05-04-2021 20:18:39.111 [Gauge] [ERROR] notif #0: initialized: current runner is not compatible with gauge LSP

PS S:\Git-TVL-Mar25\TVLQA\gauge-pom> gauge -v
Gauge version: 1.1.8
Commit Hash: b1501f4

Plugins
-------
html-report (4.0.12)
python (0.3.15)     
screenshot (0.0.1)  
xml-report (0.2.3)  

manifest.json 
`{
  "Language": "python",
  "Plugins": [
    "html-report",
    "xml-report"
  ]
}`
I initialized a new gauge python project and it works fine and gauge launches successfully, but the framework setup from git does not work on my windows pc. is there any changes that has to be made in any of .json files. I have been trying to resolve this issue by different combinations but have been unsuccessful.

your help is much appreciated.

**Comment by sriv** [comment]
hi @2vireddy  - can you please check if there are any errors reported in `gauge.log` or `lsp.log` files?

**Reply by 2vireddy** [reply]
@sriv the error in lsp.log is `[Gauge] [ERROR] notif #0: initialized: current runner is not compatible with gauge LSP` for the framework setup downloaded from github, but on creating a new gauge project using command palette this error is not displayed in lsp.log of new project. I tried uninstalling gauge and reinstalling gauge , uninstalling python language plugin and reinstalling python language plugin, nothing seems to be working and getting the same error.

**Reply by sriv** [reply]
Is there anything in `gauge.log`? 

There's a bug where the incorrect error is surfaced via LSP. The root cause could be something to do with the environment or setup (python version, pip installation, pip requirements missing, or even incorrect file encoding/syntax errors). `gauge.log` should have more details, I believe

**Reply by 2vireddy** [reply]
gauge.log did not have any info. i will enable debug in gauge settings and see if there is any error messages in gauge.log

**Reply by sriv** [reply]
sure, also please check if `gauge run specs` works for this project.

**Comment by Immanuel-Codoid** [comment]
I'm also facing the same issue
Please find below logs

`Gauge version: 1.1.8
Commit Hash: b1501f4

Plugins
-------
html-report (4.0.12)
python (0.3.15)     
screenshot (0.0.1)  
spectacle (0.1.4)
xml-report (0.2.3)

07-04-2021 16:08:56.505 [Gauge] [DEBUG] Gauge could not initialize. Install 'ms-python.python' extension for code insights. For more information see[Problems](command:workbench.actions.view.problems), check logs.[Troubleshooting](https://docs.gauge.org/troubleshooting.html?language=javascript&ide=vscode#gauge-could-not-initialize-for-more-information-see-problems)
Some of the gauge lsp feature will not work as expected.
07-04-2021 16:08:56.505 [Gauge] [INFO] LangServer: reading on stdin, writing on stdout
07-04-2021 16:08:56.505 [Gauge] [DEBUG] jsonrpc2: <-- notif: window/showMessage: {"type":1,"message":"Gauge could not initialize. Install 'ms-python.python' extension for code insights. For more information see[Problems](command:workbench.actions.view.problems), check logs.[Troubleshooting](https://docs.gauge.org/troubleshooting.html?language=javascript\u0026ide=vscode#gauge-could-not-initialize-for-more-information-see-problems)"}
### 07-04-2021 16:08:56.506 [Gauge] [DEBUG] jsonrpc2: --> request #0: initialize: {"processId":16280,"clientInfo":{"name":"Visual Studio Code","version":"1.55.0"},"locale":"en-us","rootPath":"e:\\Gauge\\new_gauge\\TVLQA\\gauge-pom","rootUri":"file:///e%!!(MISSING)A(MISSING)/Gauge/new_gauge/TVLQA/gauge-pom","capabilities":{"workspace":{"applyEdit":true,"workspaceEdit":{"documentChanges":true,"resourceOperations":["create","rename","delete"],"failureHandling":"textOnlyTransactional","normalizesLineEndings":true,"changeAnnotationSupport":{"groupsOnLabel":true}},"didChangeConfiguration":{"dynamicRegistration":true},"didChangeWatchedFiles":{"dynamicRegistration":true},"symbol":{"dynamicRegistration":true,"symbolKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]},"tagSupport":{"valueSet":[1]}},"codeLens":{"refreshSupport":true},"executeCommand":{"dynamicRegistration":true},"configuration":true,"workspaceFolders":true,"semanticTokens":{"refreshSupport":true},"fileOperations":{"dynamicRegistration":true,"didCreate":true,"didRename":true,"didDelete":true,"willCreate":true,"willRename":true,"willDelete":true}},"textDocument":{"publishDiagnostics":{"relatedInformation":true,"versionSupport":false,"tagSupport":{"valueSet":[1,2]},"codeDescriptionSupport":true,"dataSupport":true},"synchronization":{"dynamicRegistration":true,"willSave":true,"willSaveWaitUntil":true,"didSave":true},"completion":{"dynamicRegistration":true,"contextSupport":true,"completionItem":{"snippetSupport":true,"commitCharactersSupport":true,"documentationFormat":["markdown","plaintext"],"deprecatedSupport":true,"preselectSupport":true,"tagSupport":{"valueSet":[1]},"insertReplaceSupport":true,"resolveSupport":{"properties":["documentation","detail","additionalTextEdits"]},"insertTextModeSupport":{"valueSet":[1,2]}},"completionItemKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25]}},"hover":{"dynamicRegistration":true,"contentFormat":["markdown","plaintext"]},"signatureHelp":{"dynamicRegistration":true,"signatureInformation":{"documentationFormat":["markdown","plaintext"],"parameterInformation":{"labelOffsetSupport":true},"activeParameterSupport":true},"contextSupport":true},"definition":{"dynamicRegistration":true,"linkSupport":true},"references":{"dynamicRegistration":true},"documentHighlight":{"dynamicRegistration":true},"documentSymbol":{"dynamicRegistration":true,"symbolKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]},"hierarchicalDocumentSymbolSupport":true,"tagSupport":{"valueSet":[1]},"labelSupport":true},"codeAction":{"dynamicRegistration":true,"isPreferredSupport":true,"disabledSupport":true,"dataSupport":true,"resolveSupport":{"properties":["edit"]},"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}},"honorsChangeAnnotations":false},"codeLens":{"dynamicRegistration":true},"formatting":{"dynamicRegistration":true},"rangeFormatting":{"dynamicRegistration":true},"onTypeFormatting":{"dynamicRegistration":true},"rename":{"dynamicRegistration":true,"prepareSupport":true,"prepareSupportDefaultBehavior":1,"honorsChangeAnnotations":true},"documentLink":{"dynamicRegistration":true,"tooltipSupport":true},"typeDefinition":{"dynamicRegistration":true,"linkSupport":true},"implementation":{"dynamicRegistration":true,"linkSupport":true},"colorProvider":{"dynamicRegistration":true},"foldingRange":{"dynamicRegistration":true,"rangeLimit":5000,"lineFoldingOnly":true},"declaration":{"dynamicRegistration":true,"linkSupport":true},"selectionRange":{"dynamicRegistration":true},"callHierarchy":{"dynamicRegistration":true},"semanticTokens":{"dynamicRegistration":true,"tokenTypes":["namespace","type","class","enum","interface","struct","typeParameter","parameter","variable","property","enumMember","event","function","method","macro","keyword","modifier","comment","string","number","regexp","operator"],"tokenModifiers":["declaration","definition","readonly","static","deprecated","abstract","async","modification","documentation","defaultLibrary"],"formats":["relative"],"requests":{"range":true,"full":{"delta":true}},"multilineTokenSupport":false,"overlappingTokenSupport":false},"linkedEditingRange":{"dynamicRegistration":true}},"window":{"showMessage":{"messageActionItem":{"additionalPropertiesSupport":true}},"showDocument":{"support":true},"workDoneProgress":true},"general":{"regularExpressions":{"engine":"ECMAScript","version":"ES2020"},"markdown":{"parser":"marked","version":"1.1.0"}},"saveFiles":true},"trace":"off","workspaceFolders":[{"uri":"file:///e%!!(MISSING)A(MISSING)/Gauge/new_gauge/TVLQA/gauge-pom","name":"gauge-pom"}]}
07-04-2021 16:08:56.507 [Gauge] [DEBUG] jsonrpc2: <-- result #0: initialize: {"capabilities":{"textDocumentSync":1,"completionProvider":{"resolveProvider":true,"triggerCharacters":["*","* ","\"","\u003c",":",","]},"definitionProvider":true,"documentSymbolProvider":true,"workspaceSymbolProvider":true,"codeActionProvider":true,"codeLensProvider":{},"documentFormattingProvider":true,"renameProvider":true}}
07-04-2021 16:08:56.521 [Gauge] [DEBUG] jsonrpc2: --> notif: initialized: {}
07-04-2021 16:08:56.521 [Gauge] [DEBUG] jsonrpc2: --> request #1: gauge/getRunnerLanguage: null
07-04-2021 16:08:56.521 [Gauge] [DEBUG] jsonrpc2: <-- request #0: client/registerCapability: {"registrations":[{"id":"gauge-fileWatcher","method":"workspace/didChangeWatchedFiles","registerOptions":{"watchers":[{"globPattern":"e:/Gauge/new_gauge/TVLQA/gauge-pom/**/*{.spec,.md,.cpt}","kind":5}]}}]}
07-04-2021 16:08:56.522 [Gauge] [DEBUG] jsonrpc2: <-- result #1: gauge/getRunnerLanguage: ""
07-04-2021 16:08:56.530 [Gauge] [DEBUG] jsonrpc2: --> request #0: client/registerCapability: {"registrations":[{"id":"gauge-fileWatcher","method":"workspace/didChangeWatchedFiles","registerOptions":{"watchers":[{"globPattern":"e:/Gauge/new_gauge/TVLQA/gauge-pom/**/*{.spec,.md,.cpt}","kind":5}]}}]}
07-04-2021 16:08:56.531 [Gauge] [ERROR] notif #0: initialized: current runner is not compatible with gauge LSP`

**Reply by 2vireddy** [reply]
@sriv This is the issue we are seeing. 
`07-04-2021 16:08:56.505 [Gauge] [DEBUG] Gauge could not initialize. Install 'ms-python.python' extension for code insights. For more information seeProblems, check logs.Troubleshooting
Some of the gauge lsp feature will not work as expected.
07-04-2021 16:08:56.505 [Gauge] [INFO] LangServer: reading on stdin, writing on stdout
07-04-2021 16:08:56.505 [Gauge] [DEBUG] jsonrpc2: <-- notif: window/showMessage: {"type":1,"message":"Gauge could not initialize. Install 'ms-python.python' extension for code insights. For more information seeProblems, check logs.Troubleshooting"}

07-04-2021 16:08:56.506 [Gauge] [DEBUG] jsonrpc2: --> request #0: initialize: {"processId":16280,"clientInfo":{"name":"Visual Studio Code","version":"1.55.0"},"locale":"en-us","rootPath":"e:\Gauge\new_gauge\TVLQA\gauge-pom",**_"rootUri":"file:///e%!!(MISSING)A(MISSING)/Gauge/new_gauge/TVLQA/gauge-pom_**","capabilities":{"workspace":{"applyEdit":true,"workspaceEdit":{"documentChanges":true,"resourceOperations":`

**Comment by 2vireddy** [comment]
I guess the problem was with the python extension. so we followed this link to fix python extension error.
https://github.com/microsoft/vscode-python/issues/14179.  The issue got resolved but fix was not consistent, so we tried the fix again and it worked.
For now it is working fine. @sriv please let us know if we are headed in the right direction.

**Reply by sriv** [reply]
I don't think python extension's loading should have anything to do with gauge. It is optional, i.e you should be able to run gauge specs even without python vscode extension. I am a bit puzzled by this, to be honest.

What does the getgauge version in your requirements.txt say?

**Reply by 2vireddy** [reply]
getgauge==0.3.15. we are able to run gauge specs from command prompt but vs code gauge extension was failing and throwing the error message. run scenario and debug scenario were not displaying in the spec files. since the framework has many .py files python extension takes some time to load and if we open any gauge spec files in the loading process it was throwing the vs code gauge extension error. my analysis may be wrong. Does vs code gauge extension depend on vs code python extension. so the issue was actually with vs code gauge extension and not gauge.

**Reply by sriv** [reply]
gauge-vscode does not depend on vscode-python extension.

Which version of python are you on? Can you check what `GAUGE_PYTHON_COMMAND` property is set to in the `python.properties` file?

I've seen similar error when I had python 3 installed, and the command needed to be set to `python3`. In my shell profile, I had `alias python=python3` which meant my terminal execution for `gauge run specs` would be ok, but vscode does not run properly.

**Reply by 2vireddy** [reply]
thanks for the response @sriv , i am working on a windows pc with python version 3.9 and `GAUGE_PYTHON_COMMAND` is set to python. why does `run scenario` and `debug scenario` doesn't appear above the  scenarios in spec file when gauge vs code error is displayed

**Comment by 2vireddy** [comment]
Also gauge spec files were open when VSCode is invoked which causes gauge extension to be invoked before the python extension loads and causes the error. @sriv please correct me if i am wrong
